tests/test_polaraveraging.py

```diff
@@ -0,0 +1,225 @@
+    testing.assert_raises(ValueError, pa._compute_g_x, data[0,0], data[1], data[2,0])
+    
+    # test for input range
+    testing.assert_raises(ValueError, pa._compute_g_x, 0.1, 0.1, -3.15)
+    testing.assert_raises(ValueError, pa._compute_g_x, 0.1, 0.1, 2.*np.pi+0.1)
+    testing.assert_almost_equal(pa._compute_g_x(0.1, 0.1, 0.), pa._compute_g_x(0.1, 0.1, np.pi))
+
+    
+    # test for reasonable values
+    testing.assert_almost_equal(pa._compute_g_x(100., 0., 0.), 0.0)
+    testing.assert_almost_equal(pa._compute_g_x(0., 100., np.pi/4.), 0.0)
+    testing.assert_almost_equal(pa._compute_g_x(0., 0., 0.3), 0.)
+
+def test_compute_g_t():
+    data = np.array([[0.01, 0.02, 0.01], # g1
+                     [0.01, 0.02, 0.03], # g2
+                     [0.1, 1.2, 3.]]) # phi [rad]
+
+    # test that function works for scalar and vector input
+    # testing.assert(isinstance(float, pa._compute_g_t(*(data[:,0]))))
+    # testing.assert(isinstance(np.array, pa._compute_g_t(*data)))
+    testing.assert_equal(3, len(pa._compute_g_t(*data)))
+    testing.assert_equal(pa._compute_g_t(*(data[:,0])), pa._compute_g_t(*data)[0])
+    
+    # test same length array input
+    testing.assert_raises(ValueError, pa._compute_g_t, data[0,0], data[1], data[2])
+    testing.assert_raises(ValueError, pa._compute_g_t, data[0], data[1,0], data[2])
+    testing.assert_raises(ValueError, pa._compute_g_t, data[0], data[1], data[2,0])
+    testing.assert_raises(ValueError, pa._compute_g_t, data[0,0], data[1,0], data[2])
+    testing.assert_raises(ValueError, pa._compute_g_t, data[0], data[1,0], data[2,0])
+    testing.assert_raises(ValueError, pa._compute_g_t, data[0,0], data[1], data[2,0])
+    
+    # test for input range
+    testing.assert_raises(ValueError, pa._compute_g_t, 0.1, 0.1, -3.15)
+    testing.assert_raises(ValueError, pa._compute_g_t, 0.1, 0.1, 2.*np.pi+0.1)
+    testing.assert_almost_equal(pa._compute_g_t(0.1, 0.1, 0.), pa._compute_g_t(0.1, 0.1, np.pi))
+    
+    # test for reasonable values
+    testing.assert_almost_equal(pa._compute_g_t(100., 0., 0.), -100.0)
+    testing.assert_almost_equal(pa._compute_g_t(0., 100., np.pi/4.), -100.0)
+    testing.assert_almost_equal(pa._compute_g_t(0., 0., 0.3), 0.)
+
+    
+def test_compute_radial_averages():
+    #testing input types
+    testing.assert_raises(TypeError, pa._compute_radial_averages, radius="glue", g=10, bins=[np.arange(1.,16.)])
+    testing.assert_raises(TypeError, pa._compute_radial_averages, radius=np.arange(1.,10.), g="glue", bins=[np.arange(1.,16.)])  
+    testing.assert_raises(TypeError, pa._compute_radial_averages, radius=np.arange(1.,10.), g=np.arange(1.,10.), bins='glue') 
+
+    #want radius and g to have same number of entries
+    testing.assert_raises(TypeError, pa._compute_radial_averages, radius=np.arange(1.,10.), g=np.arange(1.,7.), bins=[np.arange(1.,16.)])
+
+    #want binning to encompass entire radial range
```

Comment:
Why are these commented? Clarify comment to "ensure we throw a warning if gals fall out of bins range" or something.